components/eam/src/chemistry/yaml/aero_model/f90_yaml/modal_aero_bcscavcoef_init_beg_yml.f90

```diff
@@ -0,0 +1,81 @@
+  integer  :: y_lchnk
+  integer,save :: n_calls=0   ! some subroutines are called multiple times in one timestep, record the number of calls
+
+  !populate YAML structure
+  yaml%lev_print = 0 !level (**remove these if generating data for a dependent subroutines**)
+  yaml%nstep_print = 0 !time step(**remove these if generating data for a dependent subroutines**)
+
+  ! set ext_str if there are multiple sets of yaml output to be write out
+  ! here gives an example that "flag" in the code can be 0, 1, or 2:
+  ! write(ext_str,'(I2)') flag
+  ! ext_str = 'flag_'//adjustl(ext_str)
+
+  y_lchnk=1
+  !YAML file input generation code- DO NOT PORT to C++
+  !print all inputs one-by-one at column "yaml%col_print"
+  yaml%col_print = icolprnt(y_lchnk) !column to write data(**remove these if generating data for a dependent subroutines**)
+  y_nstep = get_nstep() !time step (**remove these if generating data for a dependent subroutines**)
+
+  yaml%flag_print = .true. ! to write or not to write data (**remove these if generating data for a dependent subroutines**)
+!  if(yaml%col_print == y_i .and. y_nstep==yaml%nstep_print .and. y_k == yaml%lev_print) then ! if this column exists in y_lchnk
+     !<
+     !In the case of y_i or y_k are not passed as arguments:
+!     if(y_nstep==yaml%nstep_print) then
+
+     !For generating data for a dependent subroutines where "yaml" derived type is already initialized:
+     if(yaml%flag_print) then
+     !>
+
+     ! record number of calls that can output yaml file. you only need to write one set of input/output
+     n_calls = n_calls+1
+     if (n_calls==1) then ! output the first call. change this if writing out other calls
+     ! if ((n_calls==1 .and. flag==0) .or. (n_calls==3 .and. flag==1) .or. (n_calls==5 .and. flag==2)) then ! this is an example of writing multiple sets of YAML files
+
+        yaml%lchnk_print = y_lchnk !(**remove these if generating data for a dependent subroutines**)
+        yaml%flag_print  = .true.!(**remove these if generating data for a dependent subroutines**)
+
+        !open I/O yaml files (it can have an extra optional argument to pass a unique string to differentiate file names)
+        call open_files('modal_aero_bcscavcoef_init', &  !intent-in
+             unit_input, unit_output) !intent-out
+!             unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print, yaml%col_print, &
+             'modal_aero_bcscavcoef_init',yaml%nstep_print, yaml%lev_print)
+
+        !< add code for writing data here>
+        call write_1d_var(unit_input, unit_output, 'dgnum_amode',ntot_amode, dgnum_amode)!write a single variable
+        call write_1d_var(unit_input, unit_output, 'sigmag_amode',ntot_amode,sigmag_amode)
+        call write_1d_var(unit_input, unit_output, 'specdens_amode',ntot_amode, specdens_amode)
```

Comment:
`specdens_amode` has more than 4 values; [see here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/26913eb0ffd0fb42cbaa0613f4ea0acec6524d93/components/eam/src/chemistry/modal_aero/modal_aero_data.F90#L197)

Reply:
You are right. I'll correct it